hts-sys/build.rs

```diff
@@ -3,34 +3,80 @@
 // This file may not be copied, modified, or distributed
 // except according to those terms.
 
+#[cfg(feature = "serde")]
+use bindgen;
+use cc;
 use fs_utils::copy::copy_directory;
-use glob::glob;
 
 use std::env;
 use std::fs;
+use std::io::Write;
 use std::path::PathBuf;
-use std::process::Command;
-
-fn sed_htslib_makefile(out: &PathBuf, patterns: &[&str], feature: &str) {
-    for pattern in patterns {
-        if !Command::new("sed")
-            .current_dir(out.join("htslib"))
-            .arg("-i")
-            .arg("-e")
-            .arg(pattern)
-            .arg("Makefile")
-            .status()
-            .unwrap()
-            .success()
-        {
-            panic!("failed to strip {} support", feature);
-        }
-    }
-}
+
+// these need to be kept in sync with the htslib Makefile
+const FILES: &[&str] = &[
```

Comment:
I wonder if missing `hfile_s3.c` disables AWS S3 functionality? Personally I would really like to have it included in the mix, TBH.

Reply:
Since we now have htslib 1.10.2 on `master`, it should be possible to do things like consult BAM headers sitting on S3 buckets directly like this:

https://gist.github.com/brainstorm/83ab0fe91b3aaa1e73072e58319d15bb#file-s3_htslib-c-L16

Reply:
Added features for s3 and gcs.

Reply:
@brainstorm we should add some unit tests for s3 / gcs -- feel free to drop them into this PR, or we can do as a follow-up.

Reply:
Agree, testing the S3 part of that, getting the following when trying to read a header from S3 though:

```
[E::hts_open_format] Failed to open file "s3://umccr-research-dev/test.bam" : Function not implemented
```

Working on it...

Reply:
Feel free to add in the `libdeflate-sys` new release and merge it, LGTM. Looks like the S3 stuff needs a bit more attention (for my use case at least).... can be done post-merging this, TBH... so, ship it if you want/can! ;)